### PR TITLE
Aspect ratio container to resolve content layout shift (CLS) issues

### LINKED ISSUE #217: Content Layout Shift (CLS) when transitioning from placeholder blur to actual image
### Bug description

When placeholders are enabled, I am observing a CLS when the placeholder switches with the final images. Seems to happen in every instance where I've set `ratio=".75"` on this site — the issue doesn't present if I've set `ratio="1"`.

https://user-images.githubusercontent.com/122848318/233290483-3caaf15a-d6dc-4b87-9823-a8b6657d1945.mov

> The video above shows no CLS occurs when `placeholder="false"`

https://user-images.githubusercontent.com/122848318/233290468-dbae9a5a-238b-4f72-88f9-6eabc289fc86.mov

> The video above shows CLS occuring when using the standard placeholder setting.

It's entirely possible I've missed something, but thought it would be worthwhile to pose the question to see if this could be an issue within responsive-images, or my own error. Seems like an issue with the backend calculations, but I could be wrong.

With thanks,
Andrew

### How to reproduce

` {{ responsive:url alt="{alt}" ratio="0.75" glide:width="1000" }}`

Similuate a slow connection and observe CLS when the placeholder swaps out.

### Logs

_No response_

### Environment

```yaml
yaml
Statamic 3.4.7 Pro
Laravel 8.83.27
PHP 8.1.13
Stache Watcher Enabled
Static Caching Disabled
dryven/faviconator 1.0.2
spatie/statamic-responsive-images 3.1.3
statamic/ssg 1.3.0
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

runtime (new)

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by ncla**
Would fix #217 #216. Wrote this over one evening, roughly it works. Uses aspect ratio CSS hack and we add CSS now to the template.

This would be a breaking change

### CODE REVIEW

File: src/Breakpoint.php [hunk 1]
```diff
@@ -304,4 +304,19 @@ private function readImageToBase64($assetPath): string|null
 
         return 'data:' . $assetMimeType . ';base64,' . base64_encode($assetContent);
     }
+
+    // TODO: rename
```
Comment (riasvdv) [inline src/Breakpoint.php:308]: Is this still a todo?

**Comment by ncla** [comment]
> Is this something you can put behind a config? Default to disabled now so it's not a breaking change?

Sure technically we could. Do you have an opinion against releasing another breaking change version? Less major breaking releases is better, in your opinion?

**Comment by riasvdv** [comment]
Breaking changes are manual updates for people, if we can avoid them I’d rather avoid it, I can also imagine not everyone wanting this functionality, so a config flag feels better as well

we could set it to be enabled by default in the next major version

**Comment by ncla** [comment]
After some break, I have made these changes opt-in either through a hidden config setting or you can just simply copy paste the new template. The methods that are used by old template have not been changed and do not need to be changed, so both versions of the template will work. As discussed, the new template will become default with next major release (lets time it with next major Statamic version) to allow for breaking change (just in case for those projects that had not published the templates to their project).

The only remaining thing I have doubts about is if I should redo the CSS to use native `aspect-ratio` property instead. The padding-bottom CSS hack has been kinda engraved into my brain as the go-to way to do containers that maintain aspect ratio, but I am wondering if 91% browser support is enough and has enough time passed that `padding-bottom` trick is dated. https://caniuse.com/?search=aspect

**Pull request closed without merging**